monero/wallet_client.go

```diff
@@ -30,11 +35,25 @@ type WalletClient interface {
 	UnlockClient()
 	GetAccounts() (*wallet.GetAccountsResponse, error)
 	GetAddress(idx uint64) (*wallet.GetAddressResponse, error)
+	PrimaryWalletAddress() mcrypto.Address
 	GetBalance(idx uint64) (*wallet.GetBalanceResponse, error)
 	Transfer(to mcrypto.Address, accountIdx, amount uint64) (*wallet.TransferResponse, error)
 	SweepAll(to mcrypto.Address, accountIdx uint64) (*wallet.SweepAllResponse, error)
-	GenerateFromKeys(kp *mcrypto.PrivateKeyPair, filename, password string, env common.Environment) error
-	GenerateViewOnlyWalletFromKeys(vk *mcrypto.PrivateViewKey, address mcrypto.Address, filename, password string) error
+	WaitForTransReceipt(req *WaitForReceiptRequest) (*wallet.Transfer, error)
```

Comment:
```suggestion
	WaitForTxReceipt(req *WaitForReceiptRequest) (*wallet.Transfer, error)
```

or just
```suggestion
	WaitForReceipt(req *WaitForReceiptRequest) (*wallet.Transfer, error)
```

sorry for the nit, just want to name things consistenty since we use `tx` a lot

Reply:
I'm also unsure about use of the word receipt since that's ethereum specific, maybe it could just be `WaitForTransfer` or something like that?

Reply:
It's not really a receipt in either blockchain. I originally implemented your second suggestion as I liked the orthogonality, but it causes a name conflict with the ethereum method of the same name. I'll implement the change in the next week when I convert the wallet interfaces to use a has-a relationship in the backend. Then we can have orthogonality without naming conflicts.

Reply:
sounds good!